src/workerd/api/node/node-compat-v2-test.wd-test

```diff
@@ -0,0 +1,17 @@
+using Workerd = import "/workerd/workerd.capnp";
+
+const unitTests :Workerd.Config = (
+  services = [
+    ( name = "node-compat-v2-test",
+      worker = (
+        modules = [
+          (name = "worker", esModule = embed "node-compat-v2-test.js"),
+          (name = "node:fs", esModule = "export default 1"),
+          (name = "node:path", esModule = "export default 2"),
+        ],
+        compatibilityDate = "2024-05-01",
+        compatibilityFlags = ["nodejs_compat_v2", "experimental"]
```

Comment:
is it sufficient to just add `"experimental"` into copatiblityFlags array? I tried this a modified version of one of the existing samples and got the following:

```
bazel-bin/src/workerd/server/workerd serve ./samples/helloworld_esm/config.capnp 
service main: The compatibility flag experimental is experimental and may break or be removed in a future version of workerd. To use this flag, you must pass --experimental on the command line.
```

only `bazel-bin/src/workerd/server/workerd serve ./samples/helloworld_esm/config.capnp --experimental` worked for me.

Reply:
my config.capnp looked like this:
```
const helloWorld :Workerd.Worker = (
  modules = [
    (name = "worker", esModule = embed "worker.js")
  ],
  compatibilityDate = "2024-05-12",
  compatibilityFlags = ["nodejs_compat", "nodejs_compat_v2", "experimental"]
);

```

Reply:
The `--experimental` label is there only while this is being evaluated. I plan to remove it before landing if we do decide to move forward on this. I'd like to make sure we get input from @irvinebroque and @mikenomitch

Reply:
I think I didn't make myself clear, what I'm puzzled about is that you are able to enable the experimental mode via a runtime flag in the config, but for me this didn't work and I had to explicitly use `--experimental` command line flag 
instead. Is that expected?

I don't know how this stuff works, and I tried doing what you did here at first but it didn't work, so when I later saw you using this approach in this PR I was surprised.